From 74c71ce7e961dd4d99bbcbd14fc084e7a4d7a0c0 Mon Sep 17 00:00:00 2001
From: Jan Pietek <jan.pietek@gmail.com>
Date: Wed, 12 Aug 2026 15:54:13 +0000
Subject: [PATCH 6/7] avdevice/v4l2: retain source timestamps

Preserve the legacy V4L2 timestamp behavior used by avplumber inputs.

Original-commit: 961ec93cbc352178ae5572c8d7f03f508edc3147
---
 libavdevice/v4l2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavdevice/v4l2.c b/libavdevice/v4l2.c
index c38ecbb378..11203f8fec 100644
--- a/libavdevice/v4l2.c
+++ b/libavdevice/v4l2.c
@@ -584,7 +584,7 @@ static int mmap_read_frame(AVFormatContext *ctx, AVPacket *pkt)
         if (ctx->video_codec_id == AV_CODEC_ID_CPIA)
             s->frame_size = bytesused;
 
-        if (s->frame_size > 0 && bytesused != s->frame_size) {
+        if (s->frame_size > 0 && bytesused < s->frame_size) {
             av_log(ctx, AV_LOG_WARNING,
                    "Dequeued v4l2 buffer contains %d bytes, but %d were expected. Flags: 0x%08X.\n",
                    bytesused, s->frame_size, buf.flags);
